src/wasm/literal.cpp

```diff
@@ -432,6 +432,15 @@ bool Literal::operator==(const Literal& other) const {
       assert(func.is() && other.func.is());
       return func == other.func;
     }
+    if (type.isString()) {
+      if (!gcData && !other.gcData) {
+        return true;
+      }
+      if (!gcData || !other.gcData) {
+        return false;
+      }
+      return gcData->values == other.gcData->values;
+    }
```

Comment:
Null `gcData` is only allowed when the type is a reference to bottom, so we don't need to handle it here.

```suggestion
    if (type.isString()) {
      return gcData->values == other.gcData->values;
    }
```

Reply:
Good catch!